src/components/TopBar.vue

```diff
@@ -17,48 +18,90 @@
           },
         ]"
-        @click="setActiveFile({ editor, id: file.id })"
-        @click.middle="closeFile({ editor, id: file.id })"
-      >
-        <span>{{ file.name }}</span>
-        <XIcon
-          size="16"
-          class="icon"
-          @click.stop="closeFile({ editor, id: file.id })"
-        />
-      </li>
-      <!-- Short cuts -->
-      <div
-        v-show="false"
-        v-shortkey="['alt', 'd']"
-        @shortkey="deleteFile({ id: activeFile ? activeFile.id : null })"
-      ></div>
+            @click="setActiveFile({ editor, id: file.id })"
+            @click.middle="closeFile({ editor, id: file.id })"
+            @click.right="setActiveFile({ editor, id: file.id })"
+        >
+          <span>{{ file.name }}</span>
+          <XIcon
+              size="16"
+              class="icon"
+              @click.stop="closeFile({ editor, id: file.id })"
+          />
+        </li>
+        <!-- Short cuts -->
+        <div
+            v-show="false"
+            v-shortkey="['alt', 'd']"
+            @shortkey="deleteFile({ id: activeFile ? activeFile.id : null })"
+        ></div>
+        <div
+            v-show="false"
+            v-shortkey="['alt', 'r']"
+            @shortkey="openRenameMode({ id: activeFile ? activeFile.id : null })"
+        ></div>
+      </ul>
+    </simplebar>
+    <SlideYUpTransition>
       <div
-        v-show="false"
-        v-shortkey="['alt', 'r']"
-        @shortkey="openRenameMode({ id: activeFile ? activeFile.id : null })"
-      ></div>
-    </ul>
-  </simplebar>
+          ref="contextMenu"
+          v-show="isContextMenuToggled"
+          class="context-menu"
+          v-click-outside="closeContextMenu"
+      >
+        <div class="option-item" @click="closeOtherTabs">
+          <x-icon size="14" class="icon" /> Close other tabs
+        </div>
+      </div>
+    </SlideYUpTransition>
+  </div>
+
 </template>
 
 <script>
+import { SlideYUpTransition } from "vue2-transitions";
 import { XIcon } from "vue-feather-icons";
 import { mapActions, mapGetters } from "vuex";
 
 export default {
   components: {
     XIcon,
+    SlideYUpTransition
   },
   props: {
     editor: String,
     openFiles: Array,
     activeFile: Object,
     isActive: Boolean,
   },
+  data() {
+    return {
+      isContextMenuToggled: false,
+    };
+  },
+  computed: {
+    ...mapGetters("Editor", ["getActiveFiles", "getOpenFiles"]),
+  },
   methods: {
     ...mapActions("Editor", ["closeFile", "setActiveFile"]),
     ...mapActions("Files", ["deleteFile", "openRenameMode"]),
+    closeContextMenu() {
+      this.isContextMenuToggled = false;
+    },
+    openContextMenu(e) {
+      this.isContextMenuToggled = !this.isContextMenuToggled;
+      this.$refs.contextMenu.style.top = `${e.y - 10}px`;
+      this.$refs.contextMenu.style.left = `${e.x - 320}px`;
+    },
+    closeOtherTabs() {
+      this.closeContextMenu();
+      const activeFileId = this.getActiveFiles.PRIMARY.id;
```

Comment:
Currently, we have multiple editors, so updated this to dynamically choose active editor instead of just "PRIMARY"